doc/api/events.md

```diff
@@ -646,16 +646,23 @@ Returns the current max listener value for the `EventEmitter` which is either
 set by [`emitter.setMaxListeners(n)`][] or defaults to
 [`events.defaultMaxListeners`][].
 
-### `emitter.listenerCount(eventName)`
+### `emitter.listenerCount(eventName, [listener])`
 
 <!-- YAML
 added: v3.2.0
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/46523
+    description: Added the listener argument.
 -->
 
 * `eventName` {string|symbol} The name of the event being listened for
+* `listener` {Function} The name of the listener
 * Returns: {integer}
 
 Returns the number of listeners listening to the event named `eventName`.
+If `listener` is provided, it will return `1` if the listener is found
```

Comment:
The method is called `listenerCount` so Instead of returning early I would return the actual number of matching listeners. We have a soft limit of 10 listeners per type. It shouldn't be a big overhead.

Reply:
You are right. I've changed the implementation. What about now?